nilearn/decomposition/base.py

```diff
@@ -550,11 +557,12 @@ def explained_variance(X, components, per_component=True):
             res = X - np.outer(projected_data[i],
                                components[i])
             res_var[i] = np.var(res)
+            # Free some memory
+            del res
         return np.maximum(0., 1. - res_var / full_var)
     else:
         lr = LinearRegression(fit_intercept=True)
         lr.fit(components.T, X.T)
-        res_var = X - lr.coef_.dot(components)
-        res_var **= 2
-        res_var = np.sum(res_var)
+        res = X - lr.coef_.dot(components)
+        res_var = np.var(res)
```

Comment:
For the sake of consistence, I would replace this by `_sum_of_squares(res).sum()`, and similarly replace full_var by `_sum_of_squares(X).sum()`
I guess this makes no numerical difference, but looks more consistent to me.
Do you agree ?
Best,

Reply:
Yes I agree and pushed the suggestion. The reason I was testing is because of your comment which is copy paster below. 

>I see that we use either var(X) or _sum_of_square(X), which may explain the discrepancy. Is there any reason for that ?